sdk/scheduler/src/main/java/com/mesosphere/sdk/state/GoalStateOverride.java

```diff
@@ -0,0 +1,144 @@
+         * In practice this state should only appear very briefly.
+         */
+        PENDING("PENDING"),
+
+        /**
+         * The override (or lack of override) has started processing but hasn't finished taking effect.
+         */
+        IN_PROGRESS("IN_PROGRESS"),
+
+        /**
+         * The override (or lack of override) has been committed.
+         */
+        COMPLETE("COMPLETE");
+
+        private final String serializedName;
+
+        private Progress(String serializedName) {
+            this.serializedName = serializedName;
+        }
+
+        /**
+         * The label which overrides in this state are given. For example "RUNNING" or "STOPPED". This is stored in task
+         * state storage.
+         *
+         * <p>WARNING: THIS IS STORED IN ZOOKEEPER TASK METADATA AND THEREFORE CANNOT EASILY BE CHANGED
+         */
+        public String getSerializedName() {
+            return serializedName;
+        }
+    }
+
+    /**
+     * Describes the current state of an override.
+     *
+     * The state of the override itself. Sample flow for enabling and disabling an override:
+     *
+     * <ol><li>NONE + COMPLETE (!isActive())</li>
+     * <li>STOPPED + PENDING</li>
+     * <li>STOPPED + IN_PROGRESS</li>
+     * <li>STOPPED + COMPLETE</li>
+     * <li>NONE + PENDING</li>
+     * <li>NONE + IN_PROGRESS</li>
+     * <li>NONE + COMPLETE (!isActive())</li></ol>
+     */
+    public static class Status {
+
+        /**
+         * The status of a task for which no overrides are applicable.
+         * The task is not entering, exiting, or currently in an override state.
+         */
+        public static final Status INACTIVE = new Status(GoalStateOverride.NONE, Progress.COMPLETE);
+
+        /**
+         * The target override state for this task. May be {@link GoalStateOverride#NONE} in the case of no override.
+         */
+        public final GoalStateOverride target;
+
+        /**
+         * The current state for transitioning to the {@link #target} in question. May be {@link Progress#NONE} in the
+         * case of no override being applicable.
+         */
+        public final Progress progress;
+
+        private Status(GoalStateOverride target, Progress progress) {
+            this.target = target;
+            this.progress = progress;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            return EqualsBuilder.reflectionEquals(this, o);
+        }
+
+        @Override
+        public int hashCode() {
+            return HashCodeBuilder.reflectionHashCode(this);
+        }
+    }
+
+    public Status newStatus(Progress progress) {
+        return new Status(this, progress);
+    }
+
+    private final String serializedName;
+    private final String pendingName;
+
+    private GoalStateOverride(String serializedName, String pendingName) {
+        // Validation: Overrides may not overlap with existing GoalStates
```

Comment:
Require all caps, no spaces for names as well?

Reply:
Good point, added a regex